alicloud/resource_alicloud_gpdb_instance.go

```diff
@@ -0,0 +1,272 @@
+package alicloud
+
+import (
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/gpdb"
+	"github.com/denverdino/aliyungo/common"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudGpdbInstance() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourceAlicloudGpdbInstanceRead,
+		Create: resourceAlicloudGpdbInstanceCreate,
+		Update: resourceAlicloudGpdbInstanceUpdate,
+		Delete: resourceAlicloudGpdbInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"instance_class": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"instance_group_count": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"instance_network_type": {
+				Type:         schema.TypeString,
+				ValidateFunc: validateAllowedStringValue([]string{string(common.Classic), string(common.VPC)}),
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+			},
+			"instance_charge_type": {
+				Type:         schema.TypeString,
+				ValidateFunc: validateAllowedStringValue([]string{string(PostPaid)}),
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				ValidateFunc: validateDBInstanceName,
+				Optional:     true,
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+			},
+			"vswitch_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Computed: true,
+			},
```

Comment:
I think the field `instance_network_type ` and `vpc_id` are redundant and them can be get by `vswitch_id`